src/annotation/AdulthoodValidator.java

```diff
@@ -0,0 +1,29 @@
+package annotation;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+
+public class AdulthoodValidator {
```

Comment:
Adulthood should be put on LocalDate fields (birthday date). You have to calculate if the customer is an adult or not. No need to pass parameters inside this annotation.